slick-testkit/src/main/scala/com/typesafe/slick/testkit/tests/InsertTest.scala

```diff
@@ -230,4 +230,90 @@ class InsertTest extends AsyncTest[JdbcTestDB] {
       }
     } yield ()).withPinnedSession
   }
+
+  // TODO DerbyProfile and PostgresProfile seems to have a problem in this case. check #2206 and #2207
+  def testInsertOrUpdateWithPrimaryKeyOnly: DBIOAction[Unit, NoStream, Effect.All] = if (!tdb.confName.matches("(derby|postgres).*")) {
```

Comment:
Why the need for a regex?

Reply:
`PostgresProfile` and `DerbyProfile` are having a problem in this case for now. So this match avoids that.

Reply:
I know, but why do you need a regex to check if it's derby or postgres? Can't you match an exact string, or pattern match on the class of tdb, or something else more elegant?

Reply:
Also, for postgres and derby instead of the test succeeding it should be marked "skipped" or "ignored." Check the test framework docs how to do that.

Reply:
was assuming there was a test framework... never mind then

Reply:
> Can't you match an exact string, or pattern match on the class of tdb, or something else more elegant?
 
Does it mean that pattern match is more elegant than regex? I don't understand that point, so please explain in detail.